### DISCUSSION TITLE
PlanetScale not working

### DISCUSSION THREAD

**Discussion by JakovGlavac** [planetscale/discussion#220] [Bug Reports] [answered]
Hi, I'm using planetscale with Prisma client in my production remix app. 
After inserting about 10K rows, planetscale isn't working anymore. 
In the web console everting works, but I can't figure out why it wouldn't work when connecting with connection string or using CLI.
Sometimes after a couple of minutes, it returns a result of a query. 

![image](https://user-images.githubusercontent.com/34971091/174132219-895d2f52-5d7b-4a83-bb64-a087b7cd24a1.png)

![image](https://user-images.githubusercontent.com/34971091/174132228-e692264c-b896-4c77-83e6-7180450193ca.png)

Just look at the query latency. This is not usable. 
Isn't that the whole idea of planetscale to scale?

Idk what to do, please help. And I think I should mention that I'm using the free tier.

**Comment by jgreet** [comment]
Free databases are fairly limited in resources and won't be scaled up but looking at that Query Insights screenshots you might just need to optimize or rewrite those top queries. 

Without knowing your application and having access to your data it's hard to suggest specific changes but, to start with, are the all of the fields that you're joining on in the top query indexed? Reading tens of thousands of rows per query with tables this size likely indicates that the database is performing full tables scans.

A good way to check for table scans is to examine the explain plan for an example of the query. You can get to an explain plan from within [Query Insights](https://docs.planetscale.com/concepts/query-insights#queries-list) or using [EXPLAIN or EXPLAIN ANALYZE on the command line](https://dev.mysql.com/blog-archive/mysql-explain-analyze/).

**Comment by JakovGlavac** [comment, marked as answer]
Yeah, that's true, but I messed up.
I had a stupidly big query.

In short planetscale is awesome.
